include/swift/AST/Types.h

```diff
@@ -1450,11 +1450,26 @@ class ParameterTypeFlags {
                                        : value - ParameterTypeFlags::Shared);
   }
 
+  ParameterTypeFlags withAutoClosure(bool isAutoClosure) const {
+    return ParameterTypeFlags(isAutoClosure
+                                  ? value | ParameterTypeFlags::AutoClosure
+                                  : value - ParameterTypeFlags::AutoClosure);
```

Comment:
Is this an actual int subtraction?  You can't do that safely unless you're asserting that the flag is already set in 'value'.

Reply:
This comment is still applicable.

Reply:
I overlooked that, it's not longer necessary for function metadata anyway, so I'll just remove all of the methods I've added to `ParameterTypeFlags`.

Reply:
And I've looked it up - it's actually not a int substitution, but rather an `OptionSet` operation, so it should be safe to do that.